aws-predictions/src/test/java/com/amplifyframework/predictions/aws/http/LivenessWebSocketTest.kt

```diff
@@ -317,6 +325,57 @@ internal class LivenessWebSocketTest {
         assertEquals(livenessWebSocket.getUserAgent(), "$baseline $additional")
     }
 
+    @Test
+    fun `web socket detects clock skew from server response`() {
+        val livenessWebSocket = createLivenessWebSocket()
+        mockkConstructor(WebSocket::class)
+        val socket: WebSocket = mockk()
+        livenessWebSocket.webSocket = socket
+        val sdf = SimpleDateFormat(livenessWebSocket.datePattern, Locale.getDefault())
+
+        // server responds saying time is actually 1 hour in the future
+        val oneHour = 1000 * 3600
+        val futureDate = sdf.format(Date(Date().time + oneHour))
+        val response: Response = mockk()
+        every { response.headers }.returns(Headers.headersOf("Date", futureDate))
+        every { response.header("Date") }.returns(futureDate)
+        livenessWebSocket.webSocketListener.onOpen(socket, response)
+
+        // now we should restart the websocket with an adjusted time
+        val openLatch = CountDownLatch(1)
+        val latchingListener = LatchingWebSocketResponseListener(
+            livenessWebSocket.webSocketListener,
+            openLatch = openLatch
+        )
+        livenessWebSocket.webSocketListener = latchingListener
+
+        server.enqueue(MockResponse().withWebSocketUpgrade(ServerWebSocketListener()))
+        server.start()
+        livenessWebSocket.start()
+
+        openLatch.await(3, TimeUnit.SECONDS)
+
+        assertTrue(livenessWebSocket.webSocket != null)
+        val originalRequest = livenessWebSocket.webSocket!!.request()
+
+        // make sure that followup request sends offset date
+        val sdfGMT = SimpleDateFormat("yyyyMMdd'T'HHmmss'Z'", Locale.getDefault())
+        sdfGMT.timeZone = TimeZone.getTimeZone("GMT")
+        val sentDate = originalRequest.url.queryParameter("X-Amz-Date") ?.let { sdfGMT.parse(it) }
+        val diff = abs(Date().time - sentDate?.time!!)
+        assert(oneHour - 10000 < diff && diff < oneHour + 10000)
```

Comment:
Need some clarification as to why 10 seconds is an acceptable window here.

Reply:
We don't have an exact time because it was created within the request, so looks like its just a comparison to make sure the time is close (since unit test could vary in time taken. 10 seconds is probably way higher than needed but don't see a problem with it.